nodewatcher/nodewatcher.py

```diff
@@ -18,25 +18,40 @@
 import os
 import time
 import sys
-import tempfile
 import logging
 import boto3
+from boto3.dynamodb.conditions import Attr
 import ConfigParser
 from botocore.exceptions import ClientError
 from botocore.config import Config
 import json
-import atexit
-import errno
 
 
 log = logging.getLogger(__name__)
 _DATA_DIR = "/var/run/nodewatcher/"
 _IDLETIME_FILE = _DATA_DIR + "node_idletime.json"
 _CURRENT_IDLETIME = 'current_idletime'
 
-def getConfig(instance_id):
-    log.debug('reading /etc/nodewatcher.cfg')
 
+def get_asg_name(stack_name, region, proxy_config):
+    asg_conn = boto3.client('autoscaling', region_name=region, config=proxy_config)
+    asg_name = ""
+    no_asg = True
+
+    while no_asg:
+        try:
+            r = asg_conn.describe_tags(Filters=[{'Name': 'value', 'Values': [stack_name]}])
+            asg_name = r.get('Tags')[0].get('ResourceId')
+            no_asg = False
+        except IndexError as e:
+            log.error("No asg found for cluster %s" % stack_name)
+            time.sleep(30)
```

Comment:
If the asg name is not found the node will remain up forever. We need a timeout and raise the exception to the parent.